crates/core/src/factories/core_fetch/test.rs

```diff
@@ -0,0 +1,409 @@
+use std::{sync::Arc, time::Duration};
+
+use bytes::Bytes;
+use kitsune2_api::{fetch::Fetch, id::Id, AgentId, K2Error, OpId, SpaceId};
+use rand::Rng;
+use tokio::sync::Mutex;
+
+use crate::default_builder;
+
+use super::{CoreFetch, CoreFetchConfig, Transport};
+
+#[derive(Debug)]
+pub struct MockTransport {
+    requests_sent: Arc<Mutex<Vec<(OpId, AgentId)>>>,
+    fail: bool,
+}
+
+type DynMockTransport = Arc<MockTransport>;
+
+impl MockTransport {
+    fn new(fail: bool) -> DynMockTransport {
+        Arc::new(Self {
+            requests_sent: Arc::new(Mutex::new(Vec::new())),
+            fail,
+        })
+    }
+}
+
+impl Transport for MockTransport {
+    fn send_module(
+        &self,
+        _peer: kitsune2_api::Url,
+        _space: kitsune2_api::SpaceId,
+        _module: String,
+        mut data: bytes::Bytes,
+    ) -> kitsune2_api::BoxFut<'_, kitsune2_api::K2Result<()>> {
+        Box::pin(async move {
+            let op_id_bytes = data.split_to(data.len() / 2);
+            let agent_id_bytes = data;
+            let op_id = OpId::from(op_id_bytes);
+            let agent_id = AgentId::from(agent_id_bytes);
+            self.requests_sent.lock().await.push((op_id, agent_id));
```

Comment:
do we still want to push it if it is `self.fail`? Or maybe the fail-state should be tracked in the list too?

Reply:
I'd still push it because a request was sent and that's what I want to check in the tests.